cpp/src/rolling/rolling.cu

```diff
@@ -623,5 +625,216 @@ std::unique_ptr<column> rolling_window(column_view const& input,
   }
 }
 
+std::unique_ptr<column> grouped_rolling_window(table_view const& grouping_keys,
+                                               column_view const& input,
+                                               size_type preceding_window,
+                                               size_type following_window,
+                                               size_type min_periods,
+                                               std::unique_ptr<aggregation> const& aggr,
+                                               rmm::mr::device_memory_resource* mr)
+{
+  if (input.size() == 0) return empty_like(input);
+
+  if (grouping_keys.num_columns() == 0) {
+    // No grouping_keys specified. Treat as non-grouped case.
+    return rolling_window(input, preceding_window, following_window, min_periods, aggr, mr);
+  }
+
+  CUDF_EXPECTS((grouping_keys.num_rows() == input.size()), 
+               "Size mismatch between grouping_keys and input vector.");
+
+  CUDF_EXPECTS((min_periods > 0), "min_periods must be positive");
+
+  using sort_groupby_helper = cudf::experimental::groupby::detail::sort::sort_groupby_helper;
+  sort_groupby_helper helper{grouping_keys, cudf::include_nulls::NO, cudf::sorted::YES};
+
+  auto group_offsets{helper.group_offsets()};
+  group_offsets.push_back(input.size()); // Cap the end with the number of inputs.
```

Comment:
This is not needed since #4135

Reply:
Ah, brilliant. Thank you. I'll remove this code and repeat tests.

Reply:
Tests repeated, and things look good.